packages/utils/telemetry-utils/src/utils.ts

```diff
@@ -30,3 +32,121 @@ export function logIfFalse(
 	logger.send(newEvent);
 	return false;
 }
+
+/**
+ * Wraps around an existing logger matching the {@link ITelemetryBaseLogger} interface and applies
+ * a provided callback to determine if an event should be sampled in conjunction with the 'Fluid.Telemetry.DisableSampling' logger config value.
+ * If Fluid.Telemetry.DisableSampling is set to true, all events will be unsampled, otherwise they will be sampled according to your provided event sampler callback.
+ */
+export function createSampledLogger(
+	logger: ITelemetryBaseLogger,
+	eventSampler: {
+		poll: () => boolean;
+	},
+) {
+	const monitoringContext = loggerToMonitoringContext(logger);
+	const isSamplingEnabled = !monitoringContext.config.getBoolean(
+		"Fluid.Telemetry.DisableSampling",
+	);
+
+	const sampledLogger: ITelemetryBaseLogger = {
+		send: (event: ITelemetryBaseEvent) => {
+			if (isSamplingEnabled && eventSampler.poll() === true) {
+				logger.send(event);
+			}
+		},
+	};
+
+	return sampledLogger;
+}
+/**
+ * Wraps around an existing logger matching the {@link ITelemetryLoggerExt} interface and applies
+ * a provided callback to determine if an event should be sampled in conjunction with the 'Fluid.Telemetry.DisableSampling' logger config value.
+ * If Fluid.Telemetry.DisableSampling is set to true, all events will be unsampled, otherwise they will be sampled according to your provided event sampler callback.
+ */
+export function createSampledLoggerExt(
+	logger: ITelemetryLoggerExt,
+	eventSampler: {
+		poll: () => boolean;
+	},
```

Comment:
I think this needs to be its own interface `IEventSampler` or something along those lines, so we can document `poll()` as a function that should not be called manually by the consumers, or it could mess up the counts for the sampled loggers that use it. `SystematicEventSampler` should extend it, or I'm even thinking removing it for now because the `state` bit it exposes isn't really used...?